sequential.py

```diff
@@ -0,0 +1,160 @@
+                bundle_size_validation_result = self.validate_bundle_size(one_list)
+                if bundle_size_validation_result.__len__() > 0:
+                    validation_result.extend(bundle_size_validation_result)
+                    record_id = new_record_id
+                    break
+                one_list.clear()
+                one_list.append(record)
+
+            old_log_file_name = new_log_file_name
+            old_record_id = new_record_id
+            old_serial_number = new_serial_number
+            old_record_generated_at = new_record_generated_at
+            old_ode_received_at = new_ode_received_at
+
+            if record_id == 0:
+                break
+            
+        return validation_result
+
+    def validate_bundle_size(self, sorted_record_list):
+        record_id = 0
+        first_record_id = int(sorted_record_list[0]['metadata']['serialId']['recordId'])
+        last_record_id = int(sorted_record_list[-1]['metadata']['serialId']['recordId'])
+
+        validation_result = []
+        # partial or full list?
+        if first_record_id == 1:
+            # head of a partial list?
+            if last_record_id == first_record_id:
+                # full list
+                record_num = 0
+                for record in sorted_record_list:
+                    record_num += 1
+                    bundle_size = int(record['metadata']['serialId']['bundleSize'])
+                    if sorted_record_list.len() != bundle_size:
+                        validation_result.append(ValidationResult(False, "buldleSize doesn't match number of records. Record Number: '%d' record length: '%d' != bundlSize: '%d'" % (record_num, sorted_record_list.len(), bundle_size)))
+                        record_id = int(record['metadata']['serialId']['recordId'])
+                        break
+
+                bundle_size = int(sorted_record_list[0]['metadata']['serialId']['bundleSize'])
+                if last_record_id != bundle_size:
+                    validation_result.append(ValidationResult(False, "buldleSize doesn't match the last recordId. Record Number: '%d' Last recordId: '%d' != bundlSize: '%d'" % (record_num, last_record_id, bundle_size)))
+                    record_id = last_record_id
+        else:
+            # tail of a partial list
+            record_num = 0
+            for record in sorted_record_list:
+                record_num += 1
+                bundle_size = int(record['metadata']['serialId']['bundleSize'])
+                if last_record_id != bundle_size:
+                    validation_result.append(ValidationResult(False, "buldleSize doesn't match last recordId. Record Number: '%d' last recordId: '%d' != bundleSize: '%d'" % (record_num, last_record_id, bundle_size), record))
+                    record_id = last_record_id
+                    break
+
+        return validation_result
+
+def main():
```

Comment:
Maybe we should change this to `test()` just to communicate that this is not a standalone Python module.